refactor(transformer/async-to-generator): move transform methods to `AsyncGeneratorExecutor` and make it public

Dunqing opened the pull request:
The `AsyncGeneratorExecutor` contains many transform methods which both can used in `async-to-generator` and `async-generator-functions` plugins, because their implementations are almost the same. I am still not sure where is the best place to place it.

Dunqing commented:
* **#6992** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6992?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (approved):
Looks good to me. I'm also not sure where we should put this kind of "shared functionality". Perhaps it's fine where it is, or perhaps we should make a "shared" folder for these things - I'm not sure.

Let's leave it as is for now and revisit it if things get messy as we add more of them.

overlookmotel commented:
### Merge activity

* **Oct 29, 9:53 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
refactor(transformer/async-to-generator): move transform methods to `AsyncGeneratorExecutor` and make it public (#6992)

The `AsyncGeneratorExecutor` contains many transform methods which both can used in `async-to-generator` and `async-generator-functions` plugins, because their implementations are almost the same. I am still not sure where is the best place to place it.
